test/suite/test_util18.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2021 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import codecs, filecmp
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# test_util18.py
+#   Utilities: wt printlog
+class test_util18(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_util18.a'
+    uri = 'table:' + tablename
+    logmax = 100
+    nentries = 5
+    create_params = 'key_format=S,value_format=S'
+
+    # Whether user data is redacted or printed.
+    print_user_data = [
+        ('show_user_data', dict(print_user_data=True)),
+        ('no_user_data', dict(print_user_data=False)),
+    ]
+
+    scenarios = make_scenarios(print_user_data)
+
+    def conn_config(self):
+        return 'log=(archive=false,enabled,file_max=%dK)' % self.logmax
+
+    # Populate our test table with data we can check against in the printlog output.
+    def populate(self):
+        cursor = self.session.open_cursor(self.uri, None)
+        for i in range(0, self.nentries):
+            key = 'KEY' + str(i)
+            val = 'VAL' + str(i)
+            cursor[key] = val
+        cursor.close()
+
+    # Check the given printlog file reflects the data written by 'populate'.
+    def check_populated_printlog(self, log_file, expect_keyval, expect_keyval_hex):
+        for i in range(0, self.nentries):
+            key = 'KEY' + str(i)
+            val = 'VAL' + str(i)
```

Comment:
The `KEY` and `VAL` could be globally defined at the top so that both populate and this check function always use the same thing.

Reply:
Good point, updated them to global definitions.